linera-views/src/views/collection_view.rs

```diff
@@ -586,6 +586,31 @@ where
         .await?;
         Ok(keys)
     }
+
+    /// Returns the number of entries in the collection.
+    /// ```rust
+    /// # tokio_test::block_on(async {
+    /// # use linera_views::context::{create_test_memory_context, MemoryContext};
+    /// # use linera_views::collection_view::ByteCollectionView;
+    /// # use linera_views::register_view::RegisterView;
+    /// # use linera_views::views::View;
+    /// # let context = create_test_memory_context();
+    /// let mut view: ByteCollectionView<_, RegisterView<_, String>> =
+    ///     ByteCollectionView::load(context).await.unwrap();
+    /// view.load_entry_mut(&[0, 1]).await.unwrap();
+    /// view.load_entry_mut(&[0, 2]).await.unwrap();
+    /// assert_eq!(view.count().await.unwrap(), 2);
+    /// # })
+    /// ```
+    pub async fn count(&self) -> Result<usize, ViewError> {
```

Comment:
How difficult is it to make this constant time? Should we have a combinator like `Sized` similar to `Hashed` ?

Reply:
It seems impossible for me to make this constant time:
* To know the number of written entries, we need to access the storage.
* The updates can either be `Update::Removed` or `Update::Set(T)`. So, we need to iterate over the updates. Conceivably we could switch the `BTreeMap` of updates into two maps which could help.

Regarding the `WrappedHashableContainerView`, I think you mean the Deref operator. Maybe we could do that.